Refactor

timohanke opened the pull request:
- rename matchOrder -> clearAuction
- remove indices from return value
- simplify logic
- remove market orders
- choose lower end of price range, not midpoint

timohanke committed:
Refactor

- rename matchOrder -> clearAuction
- remove indices from return value
- simplify logic
- remove market orders
- choose lower end of price range, not midpoint

AndyGura commented on test/match-orders.test.mo:
```diff
@@ -70,11 +64,9 @@ do {
+  let (volume, price) = clearAuction(asks, bids);
   assert volume == 10000;
-  assert price == 90.0;
+  assert price == 0.0;
```
How is it possible to get non-zero volume with price 0? Bidders will get tokens for free?

timohanke replied:
Discussed offline

timohanke committed:
Various Updates

- Change order of return values
- Change variables names for readability
- Update comments
- Rename test file

timohanke committed:
Catch order price 0 and infinity

timohanke commented:
Replaced by PR #6
